File: dataset_reduzido/codenet/Java/Code/s302760734.java
import java.util.Scanner;

public class Main {

	public static void main(String[] args) {
		Scanner scanner = new Scanner(System.in);
		char[] s = scanner.next().toCharArray(),t = scanner.next().toCharArray();
		boolean check = false;
		for(int i=s.length-t.length;i>=0;i--){
			boolean check2 = true;
			for(int j=0;j<t.length;j++){
				if(s[i+j] != '?' && s[i+j] != t[j]){
					check2 = false;
				}
			}
			if(check2){
				check = true;
				for(int j=0;j<t.length;j++){
					s[i+j] = t[j];
				}
			}
			if(check){
				break;
			}
		}
		if(check){
			for(int i=0;i<s.length;i++){
				if(s[i] == '?'){
					System.out.print("a");
				}else{
					System.out.print(s[i]);
				}
			}
			System.out.println();
		}else{
			System.out.println("UNRESTORABLE");
		}
	}

}

File: output/Qwen/codenet/Java/Python/s302760734.py
def restore_sequence():
    s = input()
    t = input()
    check = False
    for i in range(len(s)-len(t), -1, -1):
        check2 = True
        for j in range(len(t)):
            if s[i+j] != '?' and s[i+j] != t[j]:
                check2 = False
        if check2:
            check = True
            s list = list(s)
            for j in range(len(t)):
                s list[i+j] = t[j]
            break
    if check:
        s = ''.join(x if x != '?' else 'a' for x in s list)
        print(s)
    else:
        print("UNRESTORABLE")

restore_sequence()